parser/src/main/scala/it/pagopa/interop/commons/parser/InterfaceParser.scala

```diff
@@ -0,0 +1,25 @@
+package it.pagopa.interop.commons.parser
+
+import io.circe.Json
+
+import java.nio.charset.StandardCharsets
+import scala.util.Try
+import io.circe.jawn.{parse => parseJson}
+import io.circe.yaml.parser.{parse => parseYaml}
+
+import scala.xml.Elem
+import scala.xml.XML.loadString
+
+object InterfaceParser {
+
+  def parseOpenApi(bytes: Array[Byte]): Either[Throwable, Json] = {
+    val txt: String = new String(bytes, StandardCharsets.UTF_8)
+    parseYaml(txt) orElse parseJson(txt)
+  }
+
+  def parseSoap(bytes: Array[Byte]): Either[Throwable, Elem] = {
```

Comment:
I would call it `parseWSDL` since the other is called `parseOpenApi` and not `parseYAML`